core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -1043,64 +1109,75 @@ private boolean compileQuery(@Transient SqlCompiler compiler)
-            sqlExecutionContext.storeTelemetry(cc.getType(), Telemetry.ORIGIN_POSTGRES);
-
-            switch (cc.getType()) {
-                case CompiledQuery.SELECT:
-                    typesAndSelect = typesAndSelectPool.pop();
-                    typesAndSelect.of(cc.getRecordCursorFactory(), bindVariableService);
-                    queryTag = TAG_SELECT;
-                    LOG.debug().$("cache select [sql=").$(queryText).$(", thread=").$(Thread.currentThread().getId()).$(']').$();
-                    break;
-                case CompiledQuery.INSERT:
-                    queryTag = TAG_INSERT;
-                    typesAndInsert = typesAndInsertPool.pop();
-                    typesAndInsert.of(cc.getInsertStatement(), bindVariableService);
-                    if (bindVariableService.getIndexedVariableCount() > 0) {
-                        LOG.debug().$("cache insert [sql=").$(queryText).$(", thread=").$(Thread.currentThread().getId()).$(']').$();
-                        // we can add insert to cache right away because it is local to the connection
-                        typesAndInsertCache.put(queryText, typesAndInsert);
-                    }
-                    break;
-                case CompiledQuery.COPY_LOCAL:
-                    // uncached
-                    queryTag = TAG_COPY;
-                    sendCopyInResponse(compiler.getEngine(), cc.getTextLoader());
-                    break;
-                case CompiledQuery.SET:
-                    configureContextForSet();
-                    break;
-                case CompiledQuery.ALTER:
-                    try (QueryFuture cf = cc.execute(tempSequence)) {
-                        cf.await();
-                    }
-                default:
-                    // DDL SQL
-                    queryTag = TAG_OK;
-                    break;
-            }
+            final CompiledQuery cc = compiler.compile(queryText, sqlExecutionContext); //here
+            processCompiledQuery(compiler, cc);
+
         } else {
             isEmptyQuery = true;
         }
 
         return true;
     }
 
-    private void configureContextForSet() {
-        if (SqlKeywords.isBegin(queryText)) {
-            queryTag = TAG_BEGIN;
-            transactionState = IN_TRANSACTION;
-        } else if (SqlKeywords.isCommit(queryText)) {
-            queryTag = TAG_COMMIT;
-            if (transactionState != ERROR_TRANSACTION) {
-                transactionState = COMMIT_TRANSACTION;
-            }
-        } else if (SqlKeywords.isRollback(queryText)) {
-            queryTag = TAG_ROLLBACK;
-            transactionState = ROLLING_BACK_TRANSACTION;
-        } else {
-            queryTag = TAG_SET;
+    private void processCompiledQuery(SqlCompiler compiler, CompiledQuery cq) 
+            throws PeerDisconnectedException, PeerIsSlowToReadException, SqlException {
+        sqlExecutionContext.storeTelemetry(cq.getType(), Telemetry.ORIGIN_POSTGRES);
```

Comment:
does this line belong to `compileQuery` just before calling this method?

Reply:
I'm not sure what you mean . 
I extracted it from previous version of compileQuery method . 
Blocks can contain arbitrary type and number of queries so we've two options here : 
1. store telemetry data for each command in the block with known query type (that's what PR does)
2. store telemetry data once for the whole block with unkown query type

Reply:
indeed, option 1 is better.